cpp/ps0004/main (4).cpp

```diff
@@ -1,381 +1,32 @@
-   cout<<" L \n";
-   break;
-   
-   case 77:
-   cout<<" M \n";
-   break;
-   
-   case 78:
-   cout<<" N \n";
-   break;
-   
-   case 79:
-   cout<<" O \n";
-   break;
-   
-   case 80:
-   cout<<" P \n";
-   break;
-   
-   case 81:
-   cout<<" Q \n";
-   break;
-   
-   case 82:
-   cout<<" R \n";
-   break;
-   
-   case 83:
-   cout<<" S \n";
-   break;
-   
-   case 84:
-   cout<<" T \n";
-   break;
-   
-   case 85:
-   cout<<" U \n";
-   break;
-   
-   case 86:
-   cout<<" V \n";
-   break;
-   
-   case 87:
-   cout<<" W \n";
-   break;
-   
-   case 88:
-   cout<<" X \n";
-   break;
-   
-   case 89:
-   cout<<" Y \n";
-   break;
-   
-   case 90:
-   cout<<" Z \n";
-   break;
-   
-   case 91:
-   cout<<" [ \n";
-   break;
-   
-   case 92:
-   cout<<" | \n";
-   break;
-   
-   case 93:
-   cout<<" ] \n";
-   break;
-   
-   case 94:
-   cout<<" ^ \n";
-   break;
-   
-   case 95:
-   cout<<" _ \n";
-   break;
-   
-   case 96:
-   cout<<" . \n";
-   break;
-   
-   case 97:
-   cout<<" a \n";
-   break;
-   
-   case 98:
-   cout<<" b \n";
-   break;
-   
-   case 99:
-   cout<<" c \n";
-   break;
-   
-   case 100:
-   cout<<" d \n";
-   break;
-   
-   case 101:
-   cout<<" e \n";
-   break;
-   
-   case 102:
-   cout<<" f \n";
-   break;
-   
-   case 103:
-   cout<<" g \n";
-   break;
-   
-   case 104:
-   cout<<" h \n";
-   break;
-   
-   case 105:
-   cout<<" i \n";
-   break;
-   
-   case 106:
-   cout<<" j \n";
-   break;
-   
-   case 107:
-   cout<<" k \n";
-   break;
-   
-   case 108:
-   cout<<" l \n";
-   break;
-   
-   case 109:
-   cout<<" m \n";
-   break;
-   
-   case 110:
-   cout<<" n \n";
-   break;
-   
-   case 111:
-   cout<<" o \n";
-   break;
-   
-   case 112:
-   cout<<" p \n";
-   break;
-   
-   case 113:
-   cout<<" q \n";
-   break;
-   
-   case 114
-   cout<<" r \n";
-   break;
-   
-   case 115:
-   cout<<" s \n";
-   break;
-   
-   case 116:
-   cout<<" t \n";
-   break;
-   
-   case 117:
-   cout<<" u \n";
-   break;
-   
-   case 118:
-   cout<<" v \n";
-   break;
-   
-   case 119:
-   cout<<" w \n";
-   break;
-   
-   case 120:
-   cout<<" x \n";
-   break;
-   
-   case 121:
-   cout<<" y \n";
-   break;
-   
-   case 122:
-   cout<<" z \n";
-   break;
-   
-   default:
-   cout<<"Invalid character/s or zero value. \n";
- } 
-return 0;
-}
\ No newline at end of file
+    // checking invalidaty of user's input 1/4
+	// printing the number and the opposite printable character 2/2
+    int test = num;
+    if (num == 0) {
+        std::cout << "Invalid character/s or zero";
+    }
+    else if ((num - test) != 0)
+    {
+        std::cout << "Invalid: a floating point number!";
+    }
+    else if ((num >= 32) && (num <= 126))
+    {
+        std::cout << "the character for " << num << " is " << char(num) << "\n";
+    }
+    else if (num > 126 || num < 32)
+    {
+        std::cout << "invalid:number is out of the range! ";
+    }
+    
+    return 0;
+}
+// good work 👍 5/10
```

Comment:
- [ ] used `if` structure in a right way
- [ ] `if` structure take in consideration all invalid cases
- [ ] all expressions to check invalidity are correct
- [ ] used `std::cout` and `std::endl` or `\n` to print a different message for the 3 cases
- [x] converted the number to the opposite printable character
- [x] printed the number and the opposite printable character